### ISSUE TITLE
v0.31.6 broke some types

### ISSUE THREAD

**Issue by wasd171** [sinclairzx81/typebox#561]
Hello!

After upgrading from `0.31.5` to `0.31.6` (https://github.com/sinclairzx81/typebox/pull/560 probably related) some of our types broke:

```
const schema = Type.Object({
  data: Type.Array(
    Type.Object({ foo: Type.String() })
  ),
})
const decoded = Value.Decode(schema, res)
```

On `0.31.5` type of the `decoded` is correct: `{ data: { foo: string }[] }`
However `0.31.6` decodes to `{ data: unknown }`

**Comment by sinclairzx81** [comment]
@wasd171 Hi, thanks for reporting.

Will investigate.

**Comment by sinclairzx81** [comment]
@wasd171 Heya, Have published fix for this on `0.31.7`. 

Hey, thanks again for reporting so promptly. Seems I'd broken the decode inference path for Array on the `0.31.6` update (resolving a `Array` instead of a `TArray`). On the bright side, the fix was straight forward :)

Will close up the issue for now, but feel free to ping an issue if you run into any problems with the updated decode inference.
All the best!
S

**Comment by wasd171** [comment]
@sinclairzx81 thanks for the fix, `0.31.7` works as intended!
